doc/sphinx/language/core/assumptions.rst

```diff
@@ -71,7 +71,21 @@ the list of binders, it is expanded to a let-in definition (see
 Section :ref:`let-in`).
 
 The expression :n:`forall @ident : @type, @term` denotes the
```

Comment:
Actually, contrary to what was written there, there is a semantic restriction and this should be:
```suggestion
The expression :n:`forall @ident : @type, @type` denotes the
```

That's why I was using `@type__1` and `@type__2` in my explanations.

Reply:
Idle thought: would it make sense for `:n:` and maybe `:g:` to convert all regular spaces into nonbreaking spaces?  If so, the occasional long string might have to be split by hand.

![image](https://user-images.githubusercontent.com/1253341/122468067-13c69400-cf70-11eb-81ab-91a83985ced3.png)

Reply:
I think that would be more inconvenient than useful.